cloudant-client/src/main/java/com/cloudant/client/api/DesignDocumentManager.java

```diff
@@ -151,8 +156,21 @@ public DesignDocument get(String id, String rev) {
      */
     public Response remove(String id) {
         assertNotEmpty(id, "id");
-        DesignDocument find = db.find(DesignDocument.class, id);
-        return db.remove(find.getId(), find.getRevision());
+        String revision = null;
+        try {
+            // Get the revision ID from ETag, removing leading and trailing "
+            revision = client.executeRequest(Http.HEAD(new DatabaseURIHelper(db.getDBUri()
+            ).documentUri(id))).execute().getConnection().getHeaderField("ETag");
+        } catch (IOException ioe) {
+            throw new RuntimeException("There was an error getting revision information so the " +
+                    "design document cannot be removed.", ioe);
+        }
+        if (revision != null) {
+            revision = revision.substring(1, revision.length() - 1);
+            return db.remove(id, revision);
+        } else {
+            throw new RuntimeException("No ETag header found for design document with id " + id);
+        }

```

Comment:
Would it be helpful if a couple of tests were added for this new logic?  I can see that the end game of the logic is the same as it was before but it looks like two new conditions exist where exceptions are being thrown now.